Copy-Paste original code

Scoutress opened the pull request.

Scoutress committed:
Copy-Paste original code

ZilvinasVidmantas commented on public/images/Issivalyk-pats.jpg:
Visų nuotraukų pavadinimai turi būti vienodo  stiliaus, dažniausiai kbaba case
Nuotraukos turi būti sumažintos kokybės:
https://imagecompressor.com/
arba bandykit panašų

ZilvinasVidmantas commented on src/App.jsx:
```diff
@@ -0,0 +1,48 @@
+  );
+};
+
+const router = createBrowserRouter([
```
Iškelti routus į atskirą failą

ZilvinasVidmantas commented on src/all-services/ServicesList.jsx:
```diff
@@ -0,0 +1,36 @@
+const ServicesList = [
```
perkelti duomenis į json server

ZilvinasVidmantas commented on src/components/announcement/Announcement.jsx:
```diff
@@ -0,0 +1,33 @@
+            <span className={styles.h1Colored}>Service/Repair </span>
+            <span>Near You</span>
+          </h1>
+          <h3 className={styles.h3}>
```
po h1 turi eiti h2, tada h3 ir t.t.
negalima praleidinėti

ZilvinasVidmantas commented on src/components/announcement/Announcement.module.scss:
```diff
@@ -0,0 +1,95 @@
+  text-align: center;
+}
+
+.h1 {
```
h1 visame puslapyje tagai turėtų būti vienodo dydžio, todėl jų bendrinius stilius reiktų rašyti globaliai

ZilvinasVidmantas commented on src/components/announcement/Announcement.module.scss:
```diff
@@ -0,0 +1,95 @@
+  color: #8659d5;
+}
+
+.h3 {
```
tas pats kaip ir h1

ZilvinasVidmantas commented on src/components/announcement/Announcement.jsx:
```diff
@@ -0,0 +1,33 @@
+          </h3>
+        </div>
+        <div className={styles.searchContainer}>
+          <input
```
input;ą su jo stiliai reikėtų kelti į atskirą komponentą globaliuai

ZilvinasVidmantas commented on src/components/announcement/Announcement.jsx:
```diff
@@ -0,0 +1,33 @@
+            type="text"
+            placeholder="Search"
+          />
+          <button className={styles.searchBtn}>
```
Mygtukas turėtų būti globalus komponentas, netgi ir su ikonos variacija

ZilvinasVidmantas commented on src/components/category-list/CategoryList.jsx:
```diff
@@ -0,0 +1,43 @@
+import { useState, useEffect } from "react";
+
+const CategoryList = ({ selectedCategory }) => {
+  const [filteredServices, setFilteredServices] = useState([]);
```
state + effect  logiką iškelti į atskira  hooksą

ZilvinasVidmantas commented on src/components/category-list/CategoryList.jsx:
```diff
@@ -0,0 +1,43 @@
+        {filteredServices.map((service) => (
+          <Service
+            key={service.id}
+            name={service.name}
```
Išbandykite spread sintakse {...service
}

ZilvinasVidmantas commented on src/components/category-list/CategoryList.module.scss:
```diff
@@ -0,0 +1,32 @@
+@import url("https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,100;0,300;0,400;0,500;0,700;0,900;1,100;1,300;1,400;1,500&display=swap");
```
šriftai įtraukiamo globaliai

ZilvinasVidmantas commented on src/components/category-title/CategoryTitle.jsx:
```diff
@@ -0,0 +1,12 @@
+import PropTypes from "prop-types";
+import styles from "./CategoryTitle.module.scss";
+
+const CategoryTitle = ({ text }) => {
```
išnaudokite children prop'ą, į kurį patenka turinys, apgaubiamas komponento atidarymu ir uždarymu

ZilvinasVidmantas commented on src/components/category/Category.jsx:
```diff
@@ -0,0 +1,26 @@
+    <li>
+      <Link
+        to={link}
+        className={`${styles.button} ${isActive ? styles.active : ""}`}
```
naudokite clsx bibliotiką ir suspaprastinkite salyginius stilius

ZilvinasVidmantas commented on src/components/category/Category.module.scss:
```diff
@@ -0,0 +1,48 @@
+.button {
```
mygtukas turi būti globalus komponentas

ZilvinasVidmantas commented on src/components/header/Header.jsx:
```diff
@@ -0,0 +1,44 @@
+      <div className={styles.navbar}>
+        <nav>
+          <ul className={styles.ul}>
+            <li className={styles.button}>
```
kiekvienas są`rašo elementas gali būt atskiras lokalus komponentas

ZilvinasVidmantas commented on src/components/header/Header.jsx:
```diff
@@ -0,0 +1,44 @@
+        </nav>
+      </div>
+      <div>
+        <button className={styles.loginBtn}>
```
GLOBALUS komponentas - mygtukas

ZilvinasVidmantas commented on src/components/home-categories/HomeCategories.jsx:
```diff
@@ -0,0 +1,56 @@
+import { Link } from "react-router-dom";
+import styles from "./HomeCategories.module.scss";
+
+const categories = [
```
duomenis traukti iš json server

ZilvinasVidmantas commented on src/components/home-popular/HomePopular.jsx:
```diff
@@ -0,0 +1,27 @@
+        {popularServices.map((service) => (
+          <Service
+            key={service.id}
+            category={service.category}
```
Spreand operator in react {...props}

ZilvinasVidmantas commented on src/components/service/Service.jsx:
```diff
@@ -0,0 +1,29 @@
+        <img src={photo} alt={`${name}`} className={styles.image} />
+      </div>
+      <div className={styles.content}>
+        <h4 className={styles.category}>{category}</h4>
```
h-eader elementų eiliškumo klaida

ZilvinasVidmantas commented on src/components/sidebar/Sidebar.jsx:
```diff
@@ -0,0 +1,73 @@
+import plumbingIcon from "../../../public/icons/icons8-plumbing-50.png";
+import electricIcon from "../../../public/icons/icons8-electric-50.png";
+
+const categories = [
```
duomenlis laikyti json sreveryje

ZilvinasVidmantas commented on src/components/sidebar/Sidebar.jsx:
```diff
@@ -0,0 +1,73 @@
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  useEffect(() => {
```
const [activeCategory, setActiveCategory] = useState(selectedCategory); + useEffect(() + handleCategoryClick
iškelti į custom hooksą

atskiriamas funkcionalumas nuo atavizdavimo

ZilvinasVidmantas commented on src/components/sidebar/Sidebar.jsx:
```diff
@@ -0,0 +1,73 @@
+        {categories.map((category) => (
+          <li key={category.name}>
+            <button
+              className={`${styles.button} ${
```
clsx

ZilvinasVidmantas commented on src/pages/About.jsx:
```diff
@@ -0,0 +1,9 @@
+const About = () => {
```
jei naudojate daugybinį export, nenaudokite tam pačiam konstruktui ir export default

ZilvinasVidmantas commented on src/pages/Home.jsx:
```diff
@@ -0,0 +1,15 @@
+import Announcement from "../components/announcement/Announcement";
```
tik viename puslapyje naudojami komponentai turėtų bųti to puslapio aplanke

ZilvinasVidmantas reviewed (commented):
Labai neblogai

Scoutress committed:
Changed name style and lowered quality of photos

Scoutress committed:
Separated Root and Router from App.jsx

Scoutress committed:
Text formatting and styling changes

Fixed text heading tags.
Fixed text heading sizes.
Font added globally.

Scoutress committed:
Added JSON server and some file reorganizing

Added JSON server for future functionality.
Separated Categories and Services data from code.
Removed not used parts of code.
Remake of Sidebar functionality and navigation routes.

Scoutress committed:
Files reorganizing, remake of services sorting

Moved certain components to pages folders.
Remake of services sorting functionality.
Fixed bug, where after selecting category on main page, there was no sorting in services page.
Minor changes.

Scoutress committed:
Implemented JSON server, removed data files

Scoutress committed:
Created global button and input field

Scoutress committed:
Header buttons converted to local components

Scoutress committed:
UseState and UseEffect hooks separated from CategoryList component

Scoutress committed:
Added Spread syntax to HomePopular and CategoryList components

Scoutress committed:
Separated some hooks and added clsx

Scoutress committed:
Added service card onClick navigation to categories page

Scoutress committed:
Created Search functionality

Scoutress committed:
Added Favorite service functionality

Scoutress committed:
Created Login functionality with JSON server

Scoutress committed:
Added Register functionality with JSON server

Scoutress committed:
Fixed bug, where in Services page I needed to press twice to make category "active"
